File: src/org/usfirst/frc5293/commands/util/InlineActionCommand.java
package org.usfirst.frc5293.commands.util;

import edu.wpi.first.wpilibj.command.Subsystem;

import java.util.List;

public class InlineActionCommand extends ActionCommand {

    private final Runnable action;

    protected InlineActionCommand(Subsystem subsystem, Runnable action) {
        super(subsystem);
        this.action = action;
    }

    protected InlineActionCommand(List<Subsystem> subsystems, Runnable action) {
        super(subsystems);
        this.action = action;
    }

    @Override
    protected void action() {
        action.run();
    }
}

File: src/org/usfirst/frc5293/commands/util/InlineActionCommand.kt
package org.usfirst.frc5293.commands.util

import edu.wpi.first.wpilibj.command.Subsystem

class InlineActionCommand : ActionCommand {

    private val action: Runnable

    constructor(subsystem: Subsystem, action: Runnable) : super(subsystem) {
        this.action = action
    }

    constructor(subsystems: List<Subsystem>, action: Runnable) : super(subsystems) {
        this.action = action
    }

    override fun action() {
        action.run()
    }
}
